### PR TITLE
Fallback to PyTorch implementation for bf16 MultiScaleDeformableAtten

### PR DESCRIPTION
**Pull request by woldier**
…tion (#44)

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

`bfloat16` is useful for AMP training because it has a larger exponent range than `float16`, which can reduce the risk of overflow and `NaN` loss during training.

For Mask2Former and other models that rely on multi-scale deformable attention, supporting a bf16 fallback path would make AMP training more stable and practical, while avoiding changes to the CUDA kernel itself.

## Modification

This PR  add a dtype check in `MultiScaleDeformableAttention`.

The PR would not add bf16 support to the CUDA operator directly. Instead, when the input dtype is `torch.bfloat16`, it would use the existing PyTorch implementation as a fallback.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/vbti-development/onedl-mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/vbti-development/onedl-mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like onedl-mmdetection or onedl-mpretrain.

### CODE REVIEW

File: mmcv/ops/multi_scale_deform_attn.py [hunk 1]
```diff
@@ -367,11 +367,20 @@ def forward(self,
...
+                and value.dtype != torch.bfloat16:
             output = MultiScaleDeformableAttnFunction.apply(
                 value, spatial_shapes, level_start_index, sampling_locations,
                 attention_weights, self.im2col_step)
         else:
+            if ((IS_CUDA_AVAILABLE and value.is_cuda)
+                or (IS_MLU_AVAILABLE and value.is_mlu)
+                or (IS_MUSA_AVAILABLE and value.is_musa)
+                or (IS_NPU_AVAILABLE and value.device.type == 'npu'))\
+                and value.dtype == torch.bfloat16:
```
[the remark this answers is not in the document]
Comment (woldier) [reply mmcv/ops/multi_scale_deform_attn.py]: Fixed the logic in this section

**Review by lauriebax — APPROVED** [review]
